File: java/FactoryMethod/src/factory/USPostCode.java
package factory;

public class USPostCode implements PostCode{
    String postalCode;
    
	public USPostCode(String postalCode)
    {
    	if(!______(postalCode)) { //> 5
    		throw new IllegalArgumentException("Invalid postcode");
    	}
    	
    	this.postalCode = postalCode;
    }
	
	@Override
	public boolean ______(String postalCode) { //> 6
		return postalCode.matches("^[0-9]{5}(?:-[0-9]{4})?$");
	}

}

File: ruby/FactoryMethod/us_post_code.rb
require_relative 'postcode'

class USPostCode < PostCode
  def initialize(post_code)
    if _________(post_code) #> 4
      @post_code = post_code
    else
      raise 'Invalid post code'
    end

    puts "Creating a instance of USPostCode #{post_code}"
  end

  def _________(post_code) #> 5
    (post_code =~ /^[0-9]{5}(?:-[0-9]{4})?$/)
  end
end